Compilation failure on nightly

## Output

```text
error[E0615]: attempted to take value of method `hi` on type `&mut syntax::codemap::Span`
  --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/interpolate_idents-0.1.7/src/lib.rs:34:56
   |
34 |                                 Some(ref mut s) => { s.hi = span.hi; },
   |                                                        ^^
   |
   = help: maybe a `()` to call it is missing?
error[E0615]: attempted to take value of method `hi` on type `&syntax::codemap::Span`
  --> /home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/interpolate_idents-0.1.7/src/lib.rs:34:66
   |
34 |                                 Some(ref mut s) => { s.hi = span.hi; },
   |                                                                  ^^
   |
   = help: maybe a `()` to call it is missing?
```

## Misc

```text
$ rustc --version
rustc 1.22.0-nightly (6f87d20a7 2017-09-29)
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -31,7 +31,7 @@ fn interpolate_idents<'a>(cx: &'a mut ExtCtxt,
                     match token {
                         TokenTree::Token(ref span, Token::Ident(ref ident)) => {
                             match new_span {
-                                Some(ref mut s) => { s.hi = span.hi; },
+                                Some(ref mut s) => { *s = s.with_hi(span.hi()); },
                                 None => { new_span = Some(span.clone()); },
                             }
                             new_ident.push_str(&ident.name.as_str());

